lib/azure/armrest/virtual_machine_service.rb

```diff
@@ -20,11 +20,11 @@ def initialize(armrest_configuration, options = {})
       # as "Basic_A1", though information is included as well.
       #
       def series(location)
-        unless @@providers_hash[provider.downcase] && @@providers_hash[provider.downcase]['locations/vmSizes']
```

Comment:
Hm, I would argue that we should probably cache the providers and provider_info methods then, probably for 24 hours at least. I think that information rarely changes.